python/cudf/cudf/_lib/binops.pyx

```diff
@@ -103,7 +103,7 @@ def apply_op(lhs, rhs, out, op):
     cdef gdf_scalar* c_scalar = NULL
     cdef gdf_column* c_out = column_view_from_column(out)
 
-    if np.isscalar(lhs):
+    if np.isscalar(lhs) or lhs is None:
```

Comment:
Break the `is None` into a separate branch, and then pass `rhs.dtype` to `gdf_scalar_from_scalar`:
```
if lhs is None:
    check_gdf_compatibility(rhs)
    c_rhs = column_view_from_column(rhs)
    c_scalar = gdf_scalar_from_scalar(lhs, rhs.dtype)
    ...
```

Reply:
updated